frontend/src/scenes/cohorts/Cohort.tsx

```diff
@@ -1,199 +1,310 @@
+                                                    }}
+                                                >
+                                                    <MatchCriteriaSelector
+                                                        onCriteriaChange={(newGroup) =>
+                                                            onCriteriaChange(newGroup, group.id)
+                                                        }
+                                                        onRemove={() => onRemoveGroup(index)}
+                                                        group={group}
+                                                        hideRemove={cohort.groups.length === 1}
+                                                    />
+                                                </KeaField>
+                                                {index < cohort.groups.length - 1 && (
+                                                    <div className="stateful-badge or mt mb">OR</div>
+                                                )}
+                                            </div>
+                                        </Group>
+                                    ))}
+                                </>
+                                <span id="add" />
+                                <div style={{ marginTop: 8, marginBottom: 8 }}>
+                                    <a
+                                        href="#add"
+                                        style={{ padding: 0 }}
+                                        onClick={() => onAddGroup()}
+                                        data-attr="add-match-criteria"
+                                    >
+                                        <PlusOutlined /> Add matching criteria
+                                    </a>
+                                </div>
+                            </Col>
+                        </Row>
+                    </>
+                )}
 
-                            {submitted && !cohort.csv && (
-                                <p style={{ color: 'var(--danger)', marginTop: 16 }}>You need to upload a CSV file.</p>
+                {!isNewCohort && (
+                    <>
+                        <Divider />
+                        <div>
+                            <h3 className="l3">Persons in this cohort</h3>
+                            {cohort.is_calculating ? (
+                                <div className="cohort-recalculating flex-center">
+                                    <Spinner size="sm" style={{ marginRight: 4 }} />
+                                    We're recalculating who belongs to this cohort. This could take up to a couple of
+                                    minutes.
+                                </div>
+                            ) : (
+                                <div style={{ marginTop: 15 }}>
+                                    <Persons cohort={cohort} />
```

Comment:
Something is broken if you try to search the users in the cohort list after they're calculated. It seems to recalculate the cohort instead of searching the list.

Reply:
ah good catch, looks like hitting enter was saving the entire form with kea forms.